<?xml version="1.0" encoding="utf-8"?>
<!-- $Revision$ -->
<!-- EN-Revision: 328315 Maintainer: aeoris Status: ready -->
<!-- Reviewed: no -->

<refentry xml:id="yaf-dispatcher.catchexception" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
 <refnamediv>
  <refname>Yaf_Dispatcher::catchException</refname>
  <refpurpose>Activar/desactivar la captura de excepciones</refpurpose>
 </refnamediv>
 
 <refsect1 role="description">
  &reftitle.description;
  <methodsynopsis>
   <modifier>public</modifier> <type>Yaf_Dispatcher</type><methodname>Yaf_Dispatcher::catchException</methodname>
   <methodparam choice="opt"><type>bool</type><parameter>flag</parameter></methodparam>
  </methodsynopsis>
  <para>
   Mientras que application.dispatcher.throwException esté activado (también se puede llamar
   al método <methodname>Yaf_Dispatcher::throwException(TRUE)</methodname> para habilitarlo), Yaf lanzará una excepción
   en lugar de emitir un error cuando ocurren errores.
  </para>
  <para>   
   Entonces, si se habilita <methodname>Yaf_Dispatcher::catchException</methodname> (también se puede habilitar estableciendo
   <link linkend="configuration.yaf.dispatcher.catchexception">application.dispatcher.catchException</link>),
   todas las excepciones no capturadas lo serán por ErrorController::error si se ha definido una.
  </para>
 </refsect1>
 
 <refsect1 role="parameters">
  &reftitle.parameters;
  <variablelist>
   <varlistentry>
    <term><parameter>flag</parameter></term>
    <listitem>
     <para>
      bool
     </para>
    </listitem>
   </varlistentry>
  </variablelist>
 </refsect1>
 
 <refsect1 role="returnvalues">
  &reftitle.returnvalues;
  <para>
   
  </para>
 </refsect1>
 
 <refsect1 role="examples">
  &reftitle.examples;
  <example>
   <title>Ejemplo de <function>Yaf_Dispatcher::catchException</function></title>
   <programlisting role="php">
<![CDATA[
/* si se define una clase ErrorController como la siguiente */
<?php
class ErrorController extends Yaf_Controller_Abstract {
     /**
      * también se puede llamar a Yaf_Request_Abstract::getException para obtener
      * la excepción no capturada.
      */
     public function errorAction($excepción) {
        /* error occurs */
        switch ($excepción->getCode()) {
            case YAF_ERR_NOTFOUND_MODULE:
            case YAF_ERR_NOTFOUND_CONTROLLER:
            case YAF_ERR_NOTFOUND_ACTION:
            case YAF_ERR_NOTFOUND_VIEW:
                echo 404, ":", $excepción->getMessage();
                break;
            default :
                $message = $excepción->getMessage();
                echo 0, ":", $excepción->getMessage();
                break;
        }
     }
}
?>
]]>
   </programlisting>
   &example.outputs.similar;
   <screen>
<![CDATA[
/* ahora, si ocurre algún error, se asume el acceso a un controlador no existente (o uno mismo puede lanzar una excepción): */
404:Could not find controller script **/application/controllers/Controlador-no-existente.php
]]>
   </screen>
  </example>
 </refsect1>
 
 <refsect1 role="seealso">
  &reftitle.seealso;
  <simplelist>
   <member><methodname>Yaf_Dispatcher::throwException</methodname></member>
   <member><methodname>Yaf_Dispatcher::setErrorHandler</methodname></member>
  </simplelist>
 </refsect1>
 
</refentry>

<!-- Keep this comment at the end of the file
Local variables:
mode: sgml
sgml-omittag:t
sgml-shorttag:t
sgml-minimize-attributes:nil
sgml-always-quote-attributes:t
sgml-indent-step:1
sgml-indent-data:t
indent-tabs-mode:nil
sgml-parent-document:nil
sgml-default-dtd-file:"~/.phpdoc/manual.ced"
sgml-exposed-tags:nil
sgml-local-catalogs:nil
sgml-local-ecat-files:nil
End:
vim600: syn=xml fen fdm=syntax fdl=2 si
vim: et tw=78 syn=sgml
vi: ts=1 sw=1
-->
